plugin/trino-exchange/src/main/java/io/trino/plugin/exchange/FileSystemExchangeSink.java

```diff
@@ -207,40 +223,111 @@ private void destroy()
     {
         private static final int INSTANCE_SIZE = ClassLayout.parseClass(BufferedStorageWriter.class).instanceSize();
 
-        private final ExchangeStorageWriter storageWriter;
+        private final FileSystemExchangeStorage exchangeStorage;
+        private final URI outputDirectory;
+        private final Optional<SecretKey> secretKey;
+        private final boolean preserveRecordsOrder;
+        private final int partitionId;
         private final BufferPool bufferPool;
         private final AtomicReference<Throwable> failure;
+        private final int maxPageStorageSizeInBytes;
+        private final long maxFileSizeInBytes;
 
+        @GuardedBy("this")
+        private ExchangeStorageWriter currentWriter;
+        @GuardedBy("this")
+        private long currentFileSize;
```

Comment:
Keeping track of `currentFileSize` feels like `ExchangeStorageWriter` responsibility. Consider moving it there and exposing `long ExchangeStorageWriter.writtenBytes()`

Reply:
That's reasonable, but that will require us to deduplicate logic in `LocalFileSystemExchangeStorage`, `S3FileSystemExchangeStorage`, and in the future Azure and GCS too. So I feel it might be better to just track it here.